core/shared/src/main/scala/fs2/internal/Scope.scala

```diff
@@ -0,0 +1,295 @@
+package fs2.internal
+
+import fs2.{Catenable, async}
+import Algebra.{Resource, Token}
+import cats.effect.{Effect, Sync}
+import cats.implicits._
+import fs2.async.Ref.Change
+import fs2.internal.Scope.ScopeState
+
+import scala.collection.immutable.ListMap
+import scala.concurrent.ExecutionContext
+
+final class Scope[F[_]] private (private val id: Token, private val parent: Option[Scope[F]])(implicit F: Sync[F]) { self =>
+
+  private val state: SyncRef[F, ScopeState[F]] = SyncRef(ScopeState.initial.asInstanceOf[ScopeState[F]])
```

Comment:
Change `initial` to be polymorphic effect and remove this cast

Reply:
hm, I wanted to have this inital as val, so it won't get recreated with every new Scope acquisition. and I wanted SyncRef to be not +F? I may put the initial to be def, and put case somehow there if you think is better style ? Or would you be so kind and suggest something instead?